File: app/src/main/java/com/boardgamegeek/provider/GamesIdMechanicsIdProvider.java
package com.boardgamegeek.provider;

import android.content.ContentUris;
import android.net.Uri;

import com.boardgamegeek.provider.BggContract.Mechanics;
import com.boardgamegeek.provider.BggDatabase.GamesMechanics;
import com.boardgamegeek.util.SelectionBuilder;

public class GamesIdMechanicsIdProvider extends BaseProvider {
	final GamesIdMechanicsProvider provider = new GamesIdMechanicsProvider();

	@Override
	protected SelectionBuilder buildSimpleSelection(Uri uri) {
		final long mechanicId = ContentUris.parseId(uri);
		return provider.buildSimpleSelection(uri).whereEquals(GamesMechanics.MECHANIC_ID, mechanicId);
	}

	@Override
	protected String getPath() {
		return addIdToPath(provider.getPath());
	}

	@Override
	protected String getType(Uri uri) {
		return Mechanics.CONTENT_ITEM_TYPE;
	}
}

File: app/src/main/java/com/boardgamegeek/provider/GamesIdMechanicsIdProvider.kt
package com.boardgamegeek.provider

import android.content.ContentUris
import android.net.Uri
import com.boardgamegeek.provider.BggContract.*
import com.boardgamegeek.provider.BggDatabase.GamesMechanics
import com.boardgamegeek.provider.BggDatabase.Tables
import com.boardgamegeek.util.SelectionBuilder

class GamesIdMechanicsIdProvider : BaseProvider() {
    override fun getType(uri: Uri) = Mechanics.CONTENT_ITEM_TYPE

    override val path = "$PATH_GAMES/#/$PATH_MECHANICS/#"

    override fun buildSimpleSelection(uri: Uri): SelectionBuilder {
        val gameId = Games.getGameId(uri)
        val mechanicId = ContentUris.parseId(uri)
        return SelectionBuilder()
                .table(Tables.GAMES_MECHANICS)
                .whereEquals(GamesMechanics.GAME_ID, gameId)
                .whereEquals(GamesMechanics.MECHANIC_ID, mechanicId)
    }
}
